includes/class-wc-payments-incentives-service.php

```diff
@@ -312,22 +314,51 @@ private function get_store_context(): array {
 			// WooCommerce active for duration in seconds.
 			'active_for'   => time() - get_option( 'woocommerce_admin_install_timestamp', time() ),
 			// Whether the store has paid orders in the last 90 days.
-			'has_orders'   => ! empty(
-				wc_get_orders(
-					[
-						'status'       => [ 'wc-completed', 'wc-processing' ],
-						'date_created' => '>=' . strtotime( '-90 days' ),
-						'return'       => 'ids',
-						'limit'        => 1,
-					]
-				)
-			),
+			'has_orders'   => $has_orders,
 			// Whether the store has at least one payment gateway enabled.
 			'has_payments' => ! empty( WC()->payment_gateways()->get_available_payment_gateways() ),
 			'has_wcpay'    => $this->has_wcpay(),
 		];
 	}
 
+	/**
+	 * Get cached value of whether the store has orders.
+	 * If orders exist, cache for a week since it won't change.
+	 * If no orders, cache for an hour to check again soon.
+	 *
+	 * @return bool Whether the store has orders.
+	 */
+	public function get_cached_has_orders(): bool {
+		$cache_key    = Database_Cache::CONNECT_INCENTIVE_KEY . '_has_orders';
+		$cached_value = $this->database_cache->get( $cache_key );
+
+		if ( null !== $cached_value ) {
+			return $cached_value['data'];
+		}
+
+		$has_orders = ! empty(
+			wc_get_orders(
+				[
+					'status'       => [ 'wc-completed', 'wc-processing' ],
```

Comment:
I think we could use the constants from `OrderInternalStatus`

Reply:
Hey 👋 I see that the class is in Core, so I think that's why we opted for hard coding the values 🤔

Reply:
Yeah. The `OrderInternalStatus` class was [introduced](https://github.com/woocommerce/woocommerce/pull/52427) in Nov 2024. I don't think we can reliably jump on relying on it. This would make WooPayments incompatible with WC versions older than that.